pandas/core/strings.py

```diff
@@ -1962,11 +1935,268 @@ def cons_row(x):
                 cons = self._orig._constructor
                 return cons(result, name=name, index=index)
 
-    @copy(str_cat)
-    def cat(self, others=None, sep=None, na_rep=None):
-        data = self._orig if self._is_categorical else self._data
-        result = str_cat(data, others=others, sep=sep, na_rep=na_rep)
-        return self._wrap_result(result, use_codes=(not self._is_categorical))
+    def _get_series_list(self, others, ignore_index=False):
+        """
+        Auxiliary function for :meth:`str.cat`. Turn potentially mixed input
+        into list of Series (elements without an index must match the length of
+        the calling Series/Index).
+
+        Parameters
+        ----------
+        input : Series, DataFrame, np.ndarrary, list-like or list-like of those
+        ignore_index : boolean, default False
+            Determines whether to forcefully align with index of the caller
+
+        Returns
+        -------
+        tuple : first element: input transformed into list of Series
+            second element: Boolean whether FutureWarning should be raised
+        """
+
+        # once str.cat defaults to alignment, this function can be simplified;
+        # will not need `ignore_index` and the second boolean output anymore
+
+        from pandas import Index, Series, DataFrame
+
+        # self._orig is either Series or Index
+        idx = self._orig if isinstance(self._orig, Index) else self._orig.index
+
+        if isinstance(others, Series):
+            fut_warn = not others.index.equals(idx)
+            los = [Series(others.values, index=idx)
+                   if ignore_index and fut_warn else others]
+            return (los, fut_warn)
+        elif isinstance(others, Index):
+            fut_warn = not others.equals(idx)
+            los = [Series(others.values,
+                          index=(idx if ignore_index else others))]
+            return (los, fut_warn)
+        elif isinstance(others, DataFrame):
+            fut_warn = not others.index.equals(idx)
+            if ignore_index and fut_warn:
+                # without copy, this could change (the corresponding list
+                # element of) "others" that was passed to str.cat
+                others = others.copy()
+                others.index = idx
+            return ([others[x] for x in others], fut_warn)
+        elif isinstance(others, np.ndarray) and others.ndim == 2:
+            others = DataFrame(others, index=idx)
+            return ([others[x] for x in others], False)
+        elif is_list_like(others):
```

Comment:
can you show a simple case where this is actually useful (e.g. the nested list-of-objects). We don't do this anywhere else in pandas. We want simplicity. This is pretty magical.